### PR TITLE
Realtime database connect

### PR DESCRIPTION
**Pull request by cichuss**
Implemented the connection to real time database. The Host creates a room in the database and the friends who joined are added to the database and automatically updated in the UI.

structure of the real time database:
image

### CODE REVIEW

**Review by izabelakalenik — COMMENTED** [review]
I don't if you are checing it, but I think we shouldn't allow people to connect to the same room if they are already in it.  Also in firebase realtime database there is one room which contains the same user twice. 

I think when someone connect to some room they should be always navigating to the host screen, even when they close this screen (so when they click "Connect with user" on the menu to enter the room again, they should be immediately move to the host screen (if they have already joined some room).

**Comment by izabelakalenik** [comment]
> I don't if you are checing it, but I think we shouldn't allow people to connect to the same room if they are already in it. Also in firebase realtime database there is one room which contains the same user twice.
> 
> I think when someone connect to some room they should be always navigating to the host screen, even when they close this screen (so when they click "Connect with user" on the menu to enter the room again, they should be immediately move to the host screen (if they have already joined some room).


so checking the user id while connnecting to room is done, the rest is for us to decide

File: lib/screens/connect_screen/host/host_bloc.dart [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    if (currentUser != null) {
+      final user = await firestoreService.fetchUser(currentUser.uid);
+
+      await roomRef.set({
+        'host': user.username,
+        'participants': {currentUser.uid: user.username}
+      });
+        } else {
+      //error
```
Comment (malgosielska) [inline lib/screens/connect_screen/host/host_bloc.dart:37]: debugPrint("") would be better than a comment, ideally we should throw some custom error here, but a debugPrint is totally fine as well
besides that, if we're checking if user's authenticated on every screen, then why do wee needt this if(currentUser != null)?

File: lib/screens/connect_screen/joined/joined_screen.dart [hunk 2]
```diff
@@ -23,20 +25,39 @@ class JoinedScreenState extends State<JoinedScreen> {
...
   Widget build(BuildContext context) {
+    final localizations = AppLocalizations.of(context);
     return MainLayout(
-      child: JoinedScreenContent(
-        roomCode: widget.roomCode,
-        friends: friends,
+      child: BlocBuilder<JoinedBloc, JoinedState>(
+        builder: (context, state) {
+          if (state is JoinedInitial) {
+            return Center(child: CircularProgressIndicator());
```
Comment (malgosielska) [inline lib/screens/connect_screen/joined/joined_screen.dart:39]: FYI: this CircularProgressIndicator(color: Colors.white) will allow you to change colors:)

File: lib/screens/connect_screen/joined/joined_bloc.dart [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    final roomRef = _database.child('rooms').child(event.roomCode);
+    _participantsRef = roomRef.child('participants');
+    final currentUser = FirebaseAuth.instance.currentUser;
+    if (currentUser != null) {
+      final user = await firestoreService.fetchUser(currentUser.uid);
+
+      await _participantsRef.child(currentUser.uid).set(user.username);
+
+    } else {
+      //error
```
Comment (malgosielska) [inline lib/screens/connect_screen/joined/joined_bloc.dart:30]: same case as above

**Review by malgosielska — APPROVED** [review]

**Review by malgosielska — APPROVED** [review]
slaay

**Pull request merged**